packages/jellyfish-api-core/__tests__/category/account/sendTokensToAddress.test.ts

```diff
@@ -0,0 +1,134 @@
+    await container.generate(1)
+  }
+
+  async function createToken (address: string, symbol: string, amount: number): Promise<void> {
+    const metadata = {
+      symbol,
+      name: symbol,
+      isDAT: true,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: address
+    }
+
+    await container.waitForWalletBalanceGTE(100)
+    await container.call('createtoken', [metadata])
+    await container.generate(1)
+
+    await container.call('utxostoaccount', [{ [address]: '100@0' }])
+    await container.generate(1)
+
+    await container.call('minttokens', [`${amount.toString()}@${symbol}`])
+    await container.generate(1)
+  }
+
+  it('should create a transaction with auto select (empty source address) Pie selection mode', async () => {
+    const address = await client.wallet.getNewAddress()
+    const transactionHex = await client.account.sendTokensToAddress({}, { [address]: ['2@CAT'] })
+    await container.generate(1)
+
+    const account = await client.account.getAccount(addressC)
+
+    expect(await client.account.getAccount(address)).toStrictEqual(['2.00000000@CAT'])
+    expect(typeof transactionHex).toStrictEqual('string')
+    expect(transactionHex.length).toStrictEqual(64)
+    expect(account[0]).toStrictEqual('48.00000000@CAT')
+  })
+
+  it('should create a transaction with Crumbs selection mode', async () => {
+    const options: SendTokensOptions = {
+      selectionMode: SelectionModeType.CRUMBS
+    }
+    const address = await client.wallet.getNewAddress()
+    const transactionHex = await client.account.sendTokensToAddress({}, { [address]: ['10@DETH'] }, options)
+    await container.generate(1)
+
+    const account = await client.account.getAccount(addressA)
+
+    expect(account[2]).toStrictEqual('10.00000000@DETH')
+    expect(await client.account.getAccount(address)).toStrictEqual(['10.00000000@DETH'])
+    expect(typeof transactionHex).toStrictEqual('string')
+    expect(transactionHex.length).toStrictEqual(64)
+  })
+
+  it('should create a transaction with multiple destination address tokens', async () => {
+    const address = await client.wallet.getNewAddress()
+    const transactionHex = await client.account.sendTokensToAddress({}, { [address]: ['2@ETH', '0.1@CAT', '10@DETH'] })
+    await container.generate(1)
+
+    expect(await client.account.getAccount(address)).toStrictEqual(['0.10000000@CAT', '2.00000000@ETH', '10.00000000@DETH'])
+    expect(typeof transactionHex).toStrictEqual('string')
+    expect(transactionHex.length).toStrictEqual(64)
+  })
+
+  it('should create a transaction with source address provided', async () => {
+    const address = await client.wallet.getNewAddress()
+    const transactionHex = await client.account.sendTokensToAddress({ [addressA]: ['10@ETH'] }, { [address]: ['10@ETH'] })
+    await container.generate(1)
+
+    expect(await client.account.getAccount(address)).toStrictEqual(['10.00000000@ETH'])
+    expect(typeof transactionHex).toStrictEqual('string')
```

Comment:
Your code is not consistent. Like here you need to check the length too. Like what you did in line 104.
Advice: Try to reread your code from top to bottom a few times

Reply:
Thanks i will do that henceforth.